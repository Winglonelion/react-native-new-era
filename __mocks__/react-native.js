import { Animated as RNAnimated } from 'react-native';

export const Animated = {
  ...RNAnimated,
  parallel: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
  timing: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
};
