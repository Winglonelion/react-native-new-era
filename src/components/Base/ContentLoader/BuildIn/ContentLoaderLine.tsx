import React, { memo } from 'react';
import ContentLoader, { Rect } from '../index';
import { IContentLoaderProps } from '../ContentLoader.types';
import { LOADING_SPEED } from './config';

export interface PropTypes extends IContentLoaderProps {
  ready?: boolean;
  children: any;
  radius?: number;
  offsetX?: number;
  offsetY?: number;
  loadingWidth?: number;
  loadingHeight?: number;
}

const ContentLoaderLine: React.FC<PropTypes> = memo(props => {
  const {
    children,
    ready = false,
    radius = 4,
    offsetX = 0,
    offsetY = 0,
    loadingHeight,
    loadingWidth,
    ...rest
  } = props;

  if (ready) return children;

  return (
    <ContentLoader
      speed={LOADING_SPEED}
      width={props.width ?? 100}
      height={props.height ?? 30}
      viewBox={`0 0 ${props.width ?? 100} ${props.height ?? 30}`}
      backgroundColor="#f3f3f3"
      foregroundColor="#ecebeb"
      {...rest}>
      <Rect
        x={offsetX}
        y={offsetY}
        rx={radius}
        ry={radius}
        width={loadingWidth ?? props.width ?? 100}
        height={loadingHeight ?? props.height ?? 30}
      />
    </ContentLoader>
  );
});

export default ContentLoaderLine;
