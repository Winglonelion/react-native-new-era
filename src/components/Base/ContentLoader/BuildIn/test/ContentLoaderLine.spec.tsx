import React from 'react';
import { shallow } from 'enzyme';
import ContentLoaderLine, { PropTypes } from '../ContentLoaderLine';
import { View } from 'react-native';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<ContentLoaderLine {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('ContentLoaderLine check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({
      children: <View />,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('ContentLoaderLine should match snapshot with loadingHeight and width', () => {
    const { wrapper } = setup({
      children: <View />,
      loadingHeight: 100,
      loadingWidth: 100,
      width: 100,
      height: 100,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('ContentLoaderLine should match snapshot with ready', () => {
    const { wrapper } = setup({
      children: <View />,
      ready: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
