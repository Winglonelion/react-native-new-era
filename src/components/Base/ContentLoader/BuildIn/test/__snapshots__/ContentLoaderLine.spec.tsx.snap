// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoaderLine check snapshot ContentLoaderLine should match snapshot with loadingHeight and width 1`] = `
<ContentLoader
  backgroundColor="#f3f3f3"
  foregroundColor="#ecebeb"
  height={100}
  speed={1.5}
  viewBox="0 0 100 100"
  width={100}
>
  <Rect
    height={100}
    rx={4}
    ry={4}
    width={100}
    x={0}
    y={0}
  />
</ContentLoader>
`;

exports[`ContentLoaderLine check snapshot ContentLoaderLine should match snapshot with ready 1`] = `<View />`;

exports[`ContentLoaderLine check snapshot should match snapshot 1`] = `
<ContentLoader
  backgroundColor="#f3f3f3"
  foregroundColor="#ecebeb"
  height={30}
  speed={1.5}
  viewBox="0 0 100 30"
  width={100}
>
  <Rect
    height={30}
    rx={4}
    ry={4}
    width={100}
    x={0}
    y={0}
  />
</ContentLoader>
`;
